azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1116,6 +1124,101 @@ public void deleteService(final int executionId) throws ExecutorManagerException
+          executionId, e.getMessage());;
+      throw e;
+    }
+  }
+
+  private String getPodEvents(final int executionId) {
+    String podEvents = "Events: \n";
+    final String podName = getPodName(executionId);
+    V1EventList events = null;
+    try {
+      logger.info("Getting Pod Events for execution Id " + executionId);
+      events = this.coreV1Api.listNamespacedEvent(this.namespace, null, null,
+          null, null, null, null, null,
+          null, null);
+    } catch (final ApiException e) {
+      logger.error("ExecId: {}, Unable to get events for Pod. Msg: {} ",
+          executionId, e.getResponseBody());
+      return "";
+    }
+    for(V1Event event : events.getItems()) {
+      if (event.getInvolvedObject().getKind().equals("Pod") && event.getInvolvedObject().getName().equals(podName)) {
+        podEvents += formatPodEvent(event);
+      }
+    }
+    if(podEvents.equals("Events: \n")) {
+      podEvents += "\t THERE ARE NO POD EVENTS \n";
+    }
+    return podEvents;
+  }
+
+  /**
+   * This method is used to obtain the Pod Logs from the K8s Core V1 API
+   *
+   * @param executionId of pod
+   * @return String of podLogs
+   */
+  private String getPodLogs(final int executionId) {
+    final String podName = getPodName(executionId);
+    String podLogs = null;
+    try {
+      logger.info("Getting Namespaced Pod Logs for execution Id " + executionId);
+      podLogs = this.coreV1Api.readNamespacedPodLog(podName, this.namespace,
+          this.flowContainerName, false, null, null,
+          "true", null, null, 1000, null);
+    } catch (final ApiException e) {
+      logger.error("ExecId: {}, Unable to get logs for Pod. Msg: {} ",
+          executionId, e.getResponseBody());
+      return "";
+    }
+    if(StringUtils.isEmpty(podLogs)) {
+      return "Pod Logs: \n\tTHE POD LOGS ARE EMPTY";
+    }
+    return "Pod Logs: \n" + podLogs + "\nEND POD LOGS";
+  }
+
+
+  private String formatPodEvent(final V1Event event) {
+    String formatedEvent = "";
+    if(Objects.nonNull(event.getLastTimestamp()) && StringUtils.isNotEmpty(event.getLastTimestamp()
+        .toString())) {
+      formatedEvent += "\t LastTimeStamp: " + event.getLastTimestamp() + "\n";
+    }
+    else {
+      formatedEvent += "\t LastTimeStamp: " + "LastTimeStamp is null, this is a new event" + "\n";
+    }
+    if(StringUtils.isNotEmpty(event.getType())) {
+      formatedEvent += "\t Type: " + event.getType() + "\n";
+    }
+    if(StringUtils.isNotEmpty(event.getReason())) {
+      formatedEvent += "\t Reason: " + event.getReason() + "\n";
+    }
+    if(StringUtils.isNotEmpty(event.getReportingInstance())) {
+      formatedEvent += "\t ReportingInstance: " + event.getReportingInstance() + "\n";
+    }
+    if(StringUtils.isNotEmpty(event.getMessage())) {
+      formatedEvent += "\t Message: " + event.getMessage() + "\n";
+    }
+    if (formatedEvent.equals("\t LastTimeStamp: " + "LastTimeStamp is null, this is a new event" + "\n")) {
```

Comment:
Why is this check required ?

Reply:
This is the null case. Perhaps there is a better way to do this. Let me think on it.